syntax = "proto3";

package inventory;

message ItemCreated {
  string item_id = 1;
  string name = 2;
  string owner = 3;
}

message ItemCheckedIn {
  string item_id = 1;
  int32 qty = 2;
  string user_id = 3;
}

message ItemCheckedOut {
  string item_id = 1;
  int32 qty = 2;
  string user_id = 3;
}

message CheckoutRejected {
  string item_id = 1;
  int32 requested_qty = 2;
  int32 available_qty = 3;
  string reason = 4;
}
